crates/codegen/src/get_node_properties.rs

```diff
@@ -529,6 +529,16 @@ fn custom_handlers(node: &Node) -> TokenStream {
         "TypeCast" => quote! {
             tokens.push(TokenProperty::from(Token::Typecast));
         },
+        "CreateDomainStmt" => quote! {
+            tokens.push(TokenProperty::from(Token::Create));
+            tokens.push(TokenProperty::from(Token::DomainP));
+            if n.type_name.is_some() {
+                tokens.push(TokenProperty::from(Token::As));
+            }
+            if n.constraints.len() > 0 {
+                tokens.push(TokenProperty::from(Token::Check));
+            }
```

Comment:
`check` should be part of the `Constraint` node, right? its currently implement as

```rust
        "Constraint" => quote! {
            match n.contype {
                // ConstrNotnull
                2 => {
                    tokens.push(TokenProperty::from(Token::Not));
                    tokens.push(TokenProperty::from(Token::NullP));
                },
                // ConstrDefault
                3 => tokens.push(TokenProperty::from(Token::Default)),
                // ConstrCheck
                6 => tokens.push(TokenProperty::from(Token::Check)),
                // ConstrPrimary
                7 => {
                    tokens.push(TokenProperty::from(Token::Primary));
                    tokens.push(TokenProperty::from(Token::Key));
                },
                // ConstrForeign
                10 => tokens.push(TokenProperty::from(Token::References)),
                _ => panic!("Unknown Constraint {:#?}", n.contype),
            }
        },
```

similarly, `As` should be part of the `TypeName` node.